### PR TITLE
fixing figures in rnn

### PR DESCRIPTION
**Pull request by pouyaAB**
Fixing the figures in rnn documentation

### CODE REVIEW

**Comment by rizar** [comment]
I compiled it and it seems like the links from second to the first layer are missing.

**Comment by vdumoulin** [comment]
Also, why have the labels changed?

**Comment by pouyaAB** [comment]
I added the links from second to the first layer.

**Comment by rizar** [comment]
Thanks, now it looks good!

@vdumoulin , what labels do you mean?

File: docs/rnn.rst [hunk 1]
```diff
@@ -63,8 +63,8 @@ receives (figure below).
    x_1; x_2; x_3;
 
    node [shape=plaintext];
-   h_0 [label="(0, 0, 0)"]; h_1 [label="(1, 1, 1)"];
-   h_2 [label="(2, 2, 2)"]; h_3 [label="(3, 3, 3)"];
+   h_0 [label="(0, 0, 0)"]; h_1 [label="(2, 2, 2)"];
+   h_2 [label="(4, 4, 4)"]; h_3 [label="(6, 6, 6)"];
```
Comment (vdumoulin) [inline docs/rnn.rst]: @rizar labels like these. I'm not sure these changes are required, as there is a doctest that verifies that what's currently in these labels corresponds to the output that the code generates.

**Comment by pouyaAB** [comment]
The output of the code doesn't match with the lables.

**Comment by vdumoulin** [comment]
Ok, I compiled the figure and I see what you did. I agree that it is correct.

May I suggest an alternative way of correcting the figure? The fact that it is now "slanted" makes it harder to read from me. Instead, why not put an arrow between `plus_1_i` and `plus_2_i` like so:

```
diff --git a/docs/rnn.rst b/docs/rnn.rst
index 508a43f..ee85685 100644
--- a/docs/rnn.rst
+++ b/docs/rnn.rst
@@ -197,10 +197,10 @@ figure below).
    h1_0 -> plus_1_1 -> h1_1 -> plus_1_2 -> h1_2 -> plus_1_3 -> h1_3;
    h2_0 -> plus_2_1 -> h2_1 -> plus_2_2 -> h2_2 -> plus_2_3 -> h2_3;
    h2_0 -> plus_1_1; h2_1 -> plus_1_2; h2_2 -> plus_1_3;
+   plus_1_1 -> plus_2_1; plus_1_2 -> plus_2_2; plus_1_3 -> plus_2_3;

    edge [style=invis];
    h2_0 -> h1_0; h2_1 -> h1_1; h2_2 -> h1_2; h2_3 -> h1_3;
-   plus_2_1 -> plus_1_1; plus_2_2 -> plus_1_2; plus_2_3 -> plus_1_3;

    { rank=source; h2_0, h2_1, h2_2, h2_3, plus_2_1, plus_2_2, plus_2_3 }
    { rank=same; h1_0, h1_1, h1_2, h1_3, plus_1_1, plus_1_2, plus_1_3 }
```

**Comment by pouyaAB** [comment]
@vdumoulin I don't know if I understand you correctly. I think you want the diamonds to be above each other and not shifted? If we put the diamonds above each other some of the arrows will go backward which doesn't look good.

**Comment by vdumoulin** [comment]
This is the figure that it would produce:

graphviz-e7e75151f41d270a0a8abc5d0b977de656b052b1

**Comment by pouyaAB** [comment]
@vdumoulin I think this is wrong. That (4,4,4) is coming from (3,3,3)+(1,1,1). In this figure (4,4,4)is coming from (1,1,1)+(1,1,1)+(1,1,1) which is (3,3,3) and not (4,4,4)

**Comment by vdumoulin** [comment]
You're right, here's the diff that would produce the correct figure:
```
diff --git a/docs/rnn.rst b/docs/rnn.rst
index 508a43f..fb21bfc 100644
--- a/docs/rnn.rst
+++ b/docs/rnn.rst
@@ -186,9 +186,9 @@ figure below).

    node [shape=plaintext];
    h1_0 [label="(0, 0, 0)"]; h1_1 [label="(1, 1, 1)"];
-   h1_2 [label="(4, 4, 4)"]; h1_3 [label="(12, 12, 12)"];
+   h1_2 [label="(3, 3, 3)"]; h1_3 [label="(8, 8, 8)"];
    h2_0 [label="(0, 0, 0)"]; h2_1 [label="(1, 1, 1)"];
-   h2_2 [label="(3, 3, 3)"]; h2_3 [label="(8, 8, 8)"];
+   h2_2 [label="(4, 4, 4)"]; h2_3 [label="(12, 12, 12)"];

    node [shape=diamond,regular=1,label="+"];
    plus_1_1; plus_1_2; plus_1_3; plus_2_1; plus_2_2; plus_2_3;
@@ -197,10 +197,10 @@ figure below).
    h1_0 -> plus_1_1 -> h1_1 -> plus_1_2 -> h1_2 -> plus_1_3 -> h1_3;
    h2_0 -> plus_2_1 -> h2_1 -> plus_2_2 -> h2_2 -> plus_2_3 -> h2_3;
    h2_0 -> plus_1_1; h2_1 -> plus_1_2; h2_2 -> plus_1_3;
+   plus_1_1 -> plus_2_1; plus_1_2 -> plus_2_2; plus_1_3 -> plus_2_3;

    edge [style=invis];
    h2_0 -> h1_0; h2_1 -> h1_1; h2_2 -> h1_2; h2_3 -> h1_3;
-   plus_2_1 -> plus_1_1; plus_2_2 -> plus_1_2; plus_2_3 -> plus_1_3;

    { rank=source; h2_0, h2_1, h2_2, h2_3, plus_2_1, plus_2_2, plus_2_3 }
    { rank=same; h1_0, h1_1, h1_2, h1_3, plus_1_1, plus_1_2, plus_1_3 }
```

**Comment by pouyaAB** [comment]
@vdumoulin You want this right? I changed add some lines to your diff to get this.
screenshot from 2016-01-27 10 59 05

**Comment by vdumoulin** [comment]
Yes, that's what I want.

**Comment by pouyaAB** [comment]
okay I committed the changes

**Comment by vdumoulin** [comment]
Looks good from my end, thanks a lot! I'll wait for the tests to pass and then merge.

**Pull request merged**